js/text-area/textarea-autoresize.ts

```diff
@@ -19,9 +19,33 @@ function onDelegatedEvent(
 }
 
 function update_height(target: HTMLTextAreaElement) {
-  // Automatically resize the textarea to fit its content.
-  target.style.height = "auto";
-  target.style.height = target.scrollHeight + "px";
+  if (target.scrollHeight > 0) {
+    // Automatically resize the textarea to fit its content.
+    target.style.height = "auto";
+    target.style.height = target.scrollHeight + "px";
+    return;
+  }
+
+  // The textarea is not visible on the page, therefore it has a 0 scroll height.
+
+  // If we should autoresize the text area height, then we can wait for the textarea to
+  // become visible and call `update_height` again. Hopefully the scroll height is no
+  // longer 0
+
+  // Create an observer to watch for the textarea becoming visible
+  const observer = new IntersectionObserver((entries, observer) => {
+    entries.forEach((entry) => {
+      // If the entry is visible (even if it's just a single pixel)
+      if (entry.intersectionRatio > 0) {
+        // Stop observing the target
+        observer.unobserve(entry.target);
+
+        // Update the height of the textarea
+        update_height(entry.target as HTMLTextAreaElement);
+      }
+    });
+  });
+  observer.observe(target);
```

Comment:
IIRC, it's surprisingly expensive to create an observer -- is there a way we can create just one that observes all the relevant targets?

Reply:
I had figured, but I didn't know they were that bad.

Related: https://www.bennadel.com/blog/3954-intersectionobserver-api-performance-many-vs-shared-in-angular-11-0-5.htm

I converted the multiple intersection observers to a single observer iff it is needed.